src/sentry/api/endpoints/internal_beacon.py

```diff
@@ -0,0 +1,19 @@
+import logging
+
+from rest_framework.response import Response
+
+from sentry.api.base import Endpoint
+from sentry.tasks.beacon import send_beacon_metric
+
+logger = logging.getLogger("beacon")
+
+
+class InternalBeaconEndpoint(Endpoint):
+    permission_classes = ()
+
+    def post(self, request):
+        # Because this is used by the frontend, we want our frontend calls to
+        # be batched in order to reduce the number requests.
+        send_beacon_metric.delay(metrics=request.data.get("batch_data", []))
```

Comment:
Should we validate this data at all? Wondering if we have any concerns about people sending a lot of large payloads here. How are beacons stored on Sentry saas?

Reply:
Validation would be a good idea - these metrics are stored in BQ (uses `sentry.analytics`).

The payloads could have arbitrary fields, would we enforce a payload size limit?

Reply:
Ahh I wasn't sure if it was just arbitrary... should the fields be restricted to just being scalar types, or do we want to allow more complicated nested structured? If we had that restriction, we could at least say that each entry has at most X values, and restrict that + number of entries

Reply:
Yeah, I think restricting to a scalar would be fine.